concent_api/core/payments/report.py

```diff
@@ -0,0 +1,759 @@
+            warnings,
+            matched_subtask_result_with_additional_verification_payment_for_provider,
+            additional_verification_payment,
+        )
+
+    for force_payment_committed_pending_response in responses.force_payment_committed:
+        matched_settlement_payment_for_force_payment_committed = find_settlement_payment_for_force_payment_committed(
+            force_payment_committed_pending_response,
+            payments.settlement_payments,
+        )
+
+        error_was_added = check_error_cannot_match_message_with_any_payment(
+            errors,
+            matched_settlement_payment_for_force_payment_committed,
+            force_payment_committed_pending_response,
+        )
+
+        if error_was_added:
+            check_warning_deposit_claim_exists_for_missing_match(
+                warnings,
+                force_payment_committed_pending_response.subtask.subtask_id,
+                ConcentUseCase.FORCED_PAYMENT,
+            )
+
+        check_error_more_matching_payment_for_message_than_given_amount(
+            errors,
+            matched_settlement_payment_for_force_payment_committed,
+            force_payment_committed_pending_response,
+        )
+        check_error_message_value_is_different_than_payment_value(
+            errors,
+            matched_settlement_payment_for_force_payment_committed,
+            force_payment_committed_pending_response,
+        )
+        check_error_message_payment_ts_is_different_than_payment_closure_time(
+            errors,
+            matched_settlement_payment_for_force_payment_committed,
+            force_payment_committed_pending_response,
+        )
+        check_warning_force_payment_committed_indicates_that_deposit_was_too_low_to_cover_whole_cost(
+            warnings,
+            force_payment_committed_pending_response,
+        )
+
+    for forced_subtask_payment in payments.forced_subtask_payments:
+        check_error_deposit_claim_exists_for_forced_payment(
+            errors,
+            forced_subtask_payment.tx_hash,
+        )
+
+    for settlement_payment in payments.settlement_payments:
+        check_error_deposit_claim_exists_for_settlement_payment(
+            errors,
+            settlement_payment.tx_hash,
+        )
+
+    return errors, warnings
+
+
+def check_additional_verification_payment_errors_and_warning(
+    errors: List[Error],
+    warnings: List[Warnings],
+    matched_objects: List[PendingResponse],
+    additional_verification_payment: CoverAdditionalVerificationEvent,
+) -> None:
+    """ Check errors and warning related to CoverAdditionalVerificationEvent object. """
+    error_was_added = check_error_cannot_match_payment_with_any_message(
+        errors,
+        matched_objects,
+        additional_verification_payment
+    )
+
+    if error_was_added:
+        check_warning_deposit_claim_exists_for_missing_match(
+            warnings,
+            additional_verification_payment.subtask_id,
```

Comment:
I think `error_was_added` should be checked inside the function. And of course it should not be checking if there was an error but the condition that caused the error. Otherwise the check is incomplete.